src/util.rs

```diff
@@ -33,3 +35,21 @@ pub fn bytes_diff<'a, 'b>(base: &'a [u8], target: &'b [u8]) -> &'b [u8] {
         target.get_unchecked(end..)
     }
 }
+
+/// simple rewrite of golang sort.Search
+pub fn search<F>(n: usize, mut f: F) -> usize
+where
+    F: FnMut(usize) -> bool,
+{
+    let mut i = 0;
+    let mut j = n;
+    while i < j {
+        let h = (i + j) >> 1;
+        if !f(h) {
```

Comment:
Should there be a case for equal?

Reply:
This functions finds the first entry that matches predicate `f`, which only returns true or false, and is different from Rust's binary search. Generally, f will return true when current entry is equal to or greater than the target. It is possible to refactor this to use Rust's `Ordering::*`, but I'll do this later, when table implementation is complete.